src/main/java/org/mskcc/oncokb/curation/service/MainService.java

```diff
@@ -85,7 +88,31 @@ public Optional<Sequence> findSequenceByGene(ReferenceGenome referenceGenome, In
         return Optional.empty();
     }
 
+    private void enrichQuery(Alteration alteration) {
+        if (alteration.getGenes().size() > 0) {
+            alteration.setGenes(
+                alteration
+                    .getGenes()
+                    .stream()
+                    .map(gene -> {
+                        if (gene.getId() != null) {
+                            Optional<Gene> geneOptional = geneService.findOne(gene.getId());
+                            if (geneOptional.isEmpty()) {
+                                return gene;
+                            } else {
+                                return geneOptional.get();
+                            }
+                        }
+                        return gene;
+                    })
+                    .collect(Collectors.toSet())
+            );
+        }
+    }
+
     public void annotateAlteration(Alteration alteration) {
+        enrichQuery(alteration);
```

Comment:
Logic on line 147 seems to do same thing. Do we need both?

Reply:
Good call. Updated the logics.

Reply:
Sounds good otherwise LGTM